docs/user_guide/config.rst

```diff
@@ -49,19 +49,55 @@ multiple external repositories can be used as the example below illustrates for
       url: https://github.com/IAMconsortium/common-definitions.git/
   definitions:
     region:
-      repository: common-definitions
+      repository:
+        name: common-definitions
     variable:
       repositories:
-        - common-definitions
-        - legacy-definitions
+        - name: common-definitions
+        - name: legacy-definitions
   mappings:
-    repository: common-definitions
+    repository:
+      name: common-definitions
 
-The value in *definitions.region.repository* needs to reference the repository in the
-*repositories* section.
+The value in *definitions.region.repository* can be a list or a single value, needs to
+contain the ``name`` keyword and reference the repository in the *repositories* section.
 
 For model mappings the process is analogous using *mappings.repository*.
 
+Filter code lists imported from external repositories
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since importing the entirety of, for example, common-definitions is too much for most
+projects, the list can be filtered using ``include`` and ``exclude`` keywords. Under
+these keywords, lists of filters can be given that will be applied to the code list from
+the given repository.
+
+The filtering can be done by any attribute:
+
+.. code:: yaml
+
+  repositories:
+    common-definitions:
+      url: https://github.com/IAMconsortium/common-definitions.git/
+  definitions:
+    variable:
+      repository:
+        name: common-definitions
+        include:
+          - name: [Primary Energy*, Final Energy*]
+          - name: "Population*"
+            tier: 1
+        exclude:
+          - name: "Final Energy|*|*"
```

Comment:
Seems that this is subjective then. For me it was totally clear that this excludes anything level 2 and beyond.
I can see your point though about this being ambiguous

Reply:
I'll change the filtering logic to mirror that of pyam. This way we'll get a consistent API across both packages.

Reply:
FYI, I added a new filter-argument *depth* in pyam (see https://github.com/IAMconsortium/pyam/pull/886), because *level* changes behavior depending on the *variable*. string, so can lead to confusion.

I suggest to use *depth* in any examples.

Reply:
Thanks for the info. After looking at the PR for a couple of minutes and consulting the pyam documentation I think I understand the difference between the two. Happy to use depth for this then.

If I understand correctly, using _depth_ is the preferred option. If so I'd suggest to update the pyam tutorials (https://pyam-iamc.readthedocs.io/en/stable/tutorials/pyam_first_steps.html#Filtering-by-variables-and-levels) to use _depth_.
In addition, I'd also suggest removing level as an argument since it can lead to confusion to have two very similar but subtly different options. Especially without an example to illustrate the difference (at least I didn't find an example in the docs).

Reply:
Updated the example to use _depth_